src/coreComponents/constitutive/fluid/multifluid/MultiFluidBase.hpp

```diff
@@ -783,12 +783,14 @@ MultiFluidBase::KernelWrapper::
   real64 totalMolality = 0.0;
   for( integer ic = 0; ic < numComps; ++ic )
   {
+    // component weight can not be zero, checked in MultiFluidBase::postInputInitialization
     real64 const mwInv = 1.0 / m_componentMolarWeight[ic];
     compMoleFrac[ic] = composition[ic] * mwInv; // this is molality (units of mole/mass)
     dCompMoleFrac_dCompMassFrac[ic][ic] = mwInv;
     totalMolality += compMoleFrac[ic];
   }
 
+  GEOS_ERROR_IF( totalMolality < LvArray::NumericLimits< real64 >::epsilon, "Zero total molality, all component concentrations are equal to zero." );
```

Comment:
This may go bad prior to `totalMolality < LvArray::NumericLimits< real64 >::epsilon`?? If so perhaps a higher number here? Also can we have a graceful recovery somehow? For example if `totalMolality ~= 0` then that means that each `compMoleFrac[ic] ~= 0`, which means that `composition[ic] ~=0`? Should we put a limit on the minimum value of the primary variables? We are talking about void at this point?

Reply:
all `composition` can not be zero, the issue originates somewhere in the initialization where the check is missing